broadcastclient/broadcastclient.go

```diff
@@ -292,6 +293,10 @@ func (bc *BroadcastClient) connect(ctx context.Context, nextSeqNum arbutil.Messa
 		return nil, err
 	}
 	if err != nil {
+		var httpError rpc.HTTPError
+		if errors.As(err, &httpError) && httpError.StatusCode == 429 {
```

Comment:
I would first check if `ws.RejectConnectionError()`, the error emitted in `wsbroadcastserver/wsbroadcastserver.go` during upgrade can be read into `rpc.HTTPError`

Reply:
Thanks for catching this, updated to use ws.ConnectionRejectedError instead.